src/internal.c

```diff
@@ -1770,6 +1770,21 @@ int GetUint32(word32* v, const byte* buf, word32 len, word32* idx)
 }
 
 
+int GetSize(word32* v, const byte* buf, word32 len, word32* idx)
+{
+    int result;
+
+    result = GetUint32(v, buf, len, idx);
+    if (result == WS_SUCCESS) {
+        if (*v + *idx > len) {
```

Comment:
Can we adjust this here to match the previous comparisons? I know previously some of them were set up to avoid overflows and suspect this is one of the cases. The change would be "(*v > len - *idx)".